rest-api-spec/src/main/resources/rest-api-spec/test/indices.get_field_mapping/20_missing_field.yml

```diff
@@ -16,4 +16,4 @@
         index: test_index
         fields: not_existent
 
-  - match: { 'test_index.mappings': {}}
```

Comment:
I just realized this doesn't seem right -- why did the response change here? It seems like an API break to start returning a completely empty response here.

Reply:
This threw me as well, but if you look at the history it's actually restoring the behaviour that was there before `include_type_name` was added - see #37210.  There's specific logic in `RestGetFieldMappingAction` to return an empty response if the index exists but the field doesn't, but it wasn't being tripped beforehand because no `types` parameter was included in the request.  Now that `types` is gone entirely it has reverted back to the previous behaviour, but I agree it seems a bit odd - maybe we should just remove this extra logic entirely?

Reply:
I see now, thanks for the context! Looks like I subtly changed the behavior there without an explicit discussion.

I also think we should remove the extra logic. Given that we already used this new response format for a whole major release, I think it'd be confusing to change this format again in 8.0. It's also a breaking change that the user can't opt out of.

I noticed that with the current behavior we only return an empty object when a single field is provided. If the request contains two fields that are both non-existent, then we still return the format `{ "index": { "mappings": {} }`. This is true both of the current PR and the old 6.x logic. This seems quite inconsistent, and suggests to me that it's not so bad to keep the `{ "index": { "mappings": {} }` format everywhere.

Reply:
++, I'll remove that special case and re-test